x/paloma/keeper/keeper.go

```diff
@@ -152,3 +155,15 @@ func (k Keeper) CheckChainVersion(ctx sdk.Context) {
 		return
 	}
 }
+func (k *Keeper) MustGetValAddr(addr string) sdk.ValAddress {
+	defer func() {
+		if r := recover(); r != nil {
+			k.Logger(context.Background()).Error("error while getting valAddr", r)
+		}
+	}()
+	bz, err := k.AddressCodec.StringToBytes(addr)
+	if err != nil {
+		panic(err)
+	}
+	return bz
+}
```

Comment:
You don't need to panic and recover within the same function. You can just react on the error. This really should live in a util package so other modules may use it as well.

Inject codec and address, return ValAddress or empty. Probably rename to `ValAddrOrEmpty` to make intention of this function clear.

Reply:
Injecting codec and address to return valAddress with error was [added here](https://github.com/palomachain/paloma/pull/1060/commits/44f60931a2dae4143e5274f9abd8b39f88caf8b0#diff-a62f6e247365c573943d95bdb8bd694c546f42d4431e2372c59d542b279775bdR5) and later removed as you suggested.  Should we revert that util function or any modification in it required?

Reply:
I'd like to see a more comfortable way of working with this concept. The original util function you wrote was going in the right direction. Maybe we misunderstood each other.

The issue I has with it was that in [this state](https://github.com/palomachain/paloma/pull/1060/commits/44f60931a2dae4143e5274f9abd8b39f88caf8b0#diff-a62f6e247365c573943d95bdb8bd694c546f42d4431e2372c59d542b279775bdR5), it was nothing else but a wrapper - there was 0 value added. 

We agreed that an error was very unlikely to be returned, and I suggested to panic in case of an error. That whole logic was supposed to live in a util function, like this:

```go
myAddr := util.MustGetValAddr(string) // <-- will panic if the addr can't be returned
```

This is basically the way things worked with Cosmos `v0.47`, and I think it's more convenient that dealing with error handling every time.

That being said, I think I changed my mind on this. We should avoid a `panic` when we can, no matter the likelihood of occurrence, as it may result in a chain halt. That is most likely why Cosmos has removed `panic`s from their core code with `0.50` and relies on more idiomatic error handling now.

So, let's roll with them to stay closer to reference implementations and avoid `panic`. 

What does this mean for our situation? I still suggest to use a util function, that at least will ensure the address will get parsed the same way every time, and we can have an identical error message for it across modules, so it should be module and keeper agnostic, kind of like this:

```go
const ErrFailedToParseValAddress err = "failed to parse validator address from bech32"

type err string

func (e err) Error() string {
    return string(e)
}

func ValAddressFromBech32(addressCodec address.Codec, valAddr string) ([]byte, error) {
	bz, err := addressCodec.StringToBytes(valAddr)
        if err != nil {
            return nil, errors.Join(ErrFailedToParseValAddress, err)
        }
	return bz, nil
}
```